src/api/common/validate.js

```diff
@@ -24,6 +24,19 @@ function validateAddressAndSecret(obj) {
   }
 }
 
+function validateSecret(secret) {
+  if (!secret) {
+    throw error('Parameter missing: secret');
+  }
+  try {
+    if (!core.Seed.from_json(secret).is_valid()) {
```

Comment:
The nested catches are in case Seed.from_json throws somehow? Possible I guess, but I don't think it will happen intentionally (sadly), hence the `is_valid()`.